### ISSUE TITLE
ListView optional approach reveal property

### ISSUE THREAD

**Issue by touseefbsb** [Microsoft/microsoft-ui-xaml#84]
# Proposal: ListView optional approach reveal property

## Summary

May I suggest an optional boolean property to turn on approach reveal on listview? ( which will be off by default) and only those developer can turn it on who do are sure it would not cause any design problems in their app, because it is a very fluent and nice to have feature.
something like <ListView ApproachBorderReveal='true'/>

## Rationale

Recently I noticed that ListView has no approach reveal anymore, meanwhile GridView and NavigationViewList still have that, after some discussion on theme editor repo [here](https://github.com/Microsoft/fluent-xaml-theme-editor/issues/25) I discovered this was a change made in the sdk and as well as winui, and the reason for that are some design problems including **2px** border and **parallel listview items** ( as discussed in the linked issue ).
So that is why I thought it would be nice that we can have this option customize-able as needed by the developer because approach reveal is very nice to have feature for controls like ListView, so there should be a way to keep it.

Thankyou.

**Comment by kikisaints** [comment]
Thanks for logging this here, @touseefbsb.

I’m actually Imagining that this request would be a style that you could set on ListView - which would just be the original one we had before we made this change.

Thoughts?

**Comment by touseefbsb** [comment]
yeah that is a good option as far as we dont need to look at or edit the style and just set the style property of listview to a resource. or maybe a style to ItemContainerStyle property? @kikisaints

**Comment by mdtauk** [comment]
Having a style for the control - **RevealListView** for example - should be as lightweight as possible, so as to avoid any templating.  So if affordances can be added to the default template to enable it, then it makes sense.

**Comment by kikisaints** [comment]
Updated this issue to be more pitch-pleasing.

**Comment by mdtauk** [comment]
There is a plan to move away from 2 px border thickness values, so how would this affect reveal borders on the various controls?

**Comment by kikisaints** [comment]
> There is a plan to move away from 2 px border thickness values, so how would this affect reveal borders on the various controls?

I believe we're moving from 2px to 1px borders, on items that have borders today. So this *shouldn't* affect Reveal approach/border effect. Controls with borders and border Reveal will behave as they do today, post 2px changes, as far as I know. @chigy to confirm.

**Comment by chigy** [comment]
Very interesting... I just opened #1126 which originated from design team's desire to remove border reveal. We did notice some controls were no longer showing the reveal and it was actually what design wanted happen. So let's figure out the output we need. It is possible this might not be needed or maybe not exactly the same way we used to have it...
